modules/dnn/src/layers/nonzero_layer.cpp

```diff
@@ -0,0 +1,281 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the
+// top-level directory of this distribution and at http://opencv.org/license.html.
+
+#include "../precomp.hpp"
+#include "layers_common.hpp"
+#include <opencv2/dnn/shape_utils.hpp>
+#include <cstdint>
+
+namespace cv {
+namespace dnn {
+
+// ONNX NonZero operator
+// Spec: https://onnx.ai/onnx/operators/onnx__NonZero.html
+// Supported opsets: 9-18 (no attributes)
+
+namespace {
+static inline bool isNonZeroF16(uint16_t val)
+{
+    return (val & 0x7fff) != 0;
+}
+
+template <typename T, typename PT>
+static size_t computeNonZeroCountsPerStripe(const T* data, size_t total, std::vector<size_t>& nzcounts, PT isNonZero)
+{
+    const int nstripes = 16;
```

Comment:
pass nstripes as function parameter, to make sure that all stages of the algorithm use exactly the same number of stripes.